src/Surfnet/StepupGateway/SecondFactorOnlyBundle/Service/Gateway/SecondFactorRespondService.php

```diff
@@ -0,0 +1,135 @@
+     * @param ResponseFactory $responseFactory
+     * @param SecondFactorService $secondFactorService
+     * @param SecondFactorTypeService $secondFactorTypeService
+     */
+    public function __construct(
+        LoggerInterface $logger,
+        LoaResolutionService $loaResolutionService,
+        LoaAliasLookupService $loaAliasLookupService,
+        ResponseFactory $responseFactory,
+        SecondFactorService $secondFactorService,
+        SecondFactorTypeService $secondFactorTypeService
+    ) {
+        $this->logger = $logger;
+        $this->loaResolutionService = $loaResolutionService;
+        $this->loaAliasLookupService = $loaAliasLookupService;
+        $this->responseFactory = $responseFactory;
+        $this->secondFactorService = $secondFactorService;
+        $this->secondFactorTypeService = $secondFactorTypeService;
+    }
+
+
+    /**
+     * Send a SAML response back to the service provider.
+     *
+     * Second factor verification handled by SecondFactorController is
+     * finished. The user was forwarded back to this action with an internal
+     * redirect. This method sends a AuthnResponse back to the service
+     * provider in response to the AuthnRequest received in ssoAction().
+     *
+     * @param ResponseContext $responseContext
+     * @return Response
+     */
+    public function respond(ResponseContext $responseContext)
+    {
+        $this->logger->notice('Creating second-factor-only Response');
+
+        $selectedSecondFactorUuid = $responseContext->getSelectedSecondFactor();
+        if (!$selectedSecondFactorUuid) {
+            throw new InvalidSecondFactorMethodException(
+                'Cannot verify possession of an unknown second factor.'
+            );
+        }
+
+        if (!$responseContext->isSecondFactorVerified()) {
+            throw new InvalidSecondFactorMethodException(
+                'Second factor was not verified'
+            );
+        }
+
+        $secondFactor = $this->secondFactorService->findByUuid($selectedSecondFactorUuid);
+        $grantedLoa = $this->loaResolutionService
+            ->getLoaByLevel($secondFactor->getLoaLevel($this->secondFactorTypeService));
+
+        $authnContextClassRef = $this->loaAliasLookupService->findAliasByLoa($grantedLoa);
+
+        $response = $this->responseFactory->createSecondFactorOnlyResponse(
+            $responseContext->getIdentityNameId(),
+            $responseContext->getDestination(),
+            $authnContextClassRef
+        );
+
+        $this->logger->notice(sprintf(
+            'Responding to request "%s" with newly created response "%s"',
+            $responseContext->getInResponseTo(),
+            $response->getId()
+        ));
+
+        return $response;
+    }
+
+    /**
+     * Reset the state of the response
+     *
+     * @param ResponseContext $responseContext
+     */
+    public function respondReset(ResponseContext $responseContext)
```

Comment:
This methodname makes not much sense to me. Wouldn't something starting with reset not be more expressive?